### PR TITLE
chore!: bump minimum Angular version to 14

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes

This PR bumps the minimum supported Angular version from 12 to 14.

#### Commits

No commits other than below were made without a PR.

- #3861 
- #4035
- #4087
- #4116 
- #4128
- 6e9479a36a9fb3a48f068b90c277c18841220b5 (eslint fix in #4128)
- d2d12f9d730b1675ea7afd3ca904f6d6aebfd48f (code highlight fix in this PR)

#### Description of how you validated changes

Manual smoke tests + build system tests + existing CI.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/quick-start-example.angular.mdx [hunk 1]
```diff
@@ -21,10 +22,13 @@ Amplify.configure(awsconfig);
 export class AppModule {}
 ```
 
-Import Authenticator style inside `styles.css`
+Open your `angular.json` file, and add `node_modules/@aws-amplify/ui-angular/theme.css` to `styles` array in your `angular.json`. This array is located in `projects.<project-name>.architect.build.options`.
 
-```css
-@import '~@aws-amplify/ui-angular/theme.css';
+```json{3}
```
Comment (wlee221) [inline docs/src/pages/[platform]/connected-components/authenticator/quick-start-example.angular.mdx:27]:
```suggestion
```json{2}
```

File: docs/src/pages/[platform]/getting-started/installation/styles.angular.mdx [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { Tabs, TabItem } from '@aws-amplify/ui-react';
 
+Open your `angular.json` file, and add `node_modules/@aws-amplify/ui-angular/theme.css` to `styles` array in your `angular.json`. This array is located in `projects.<project-name>.architect.build.options`.
+
 ```js
```
Comment (wlee221) [inline docs/src/pages/[platform]/getting-started/installation/styles.angular.mdx:5]:
```suggestion
```json{2}
```

**Review by wlee221 — COMMENTED** [review]
Found two code snippet improvements:

**Review by hbuchel — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**